File: platform/test/integration/storage/redis/test_case.mjs
#!/usr/bin/env node

import path from 'node:path';
import { pathToFileURL } from 'node:url';

const args = process.argv.slice(2);
const readArg = (name, fallback) => {
  const idx = args.indexOf(name);
  return idx >= 0 ? args[idx + 1] : fallback;
};

const appEnv = readArg('--app-env', process.env.APP_ENV || 'dev');
const configDir = readArg('--config-dir', process.env.CONFIG_DIR);
if (!configDir) {
  console.error('Missing --config-dir');
  process.exit(2);
}

const platformRoot = path.resolve(configDir, '..', '..');
const importFile = async (absPath) => import(pathToFileURL(absPath).href);

const appYamlLoadMod = await importFile(path.join(platformRoot, 'polyglot/app_yaml_load/mjs/dist/index.js'));
const redisMod = await importFile(path.join(platformRoot, 'packages_mjs/db_connection_redis/dist/index.js'));

const { config: appConfig } = await appYamlLoadMod.loadAppYamlConfig({ configDir, appEnv });
const cfg = new redisMod.RedisConfig();

// --- Connection test ---
let connectionResult = null;
let connectionError = null;
let client = null;
try {
  client = redisMod.getRedisClient(cfg);
  const pong = await client.ping();
  connectionResult = { connected: pong === 'PONG' };
} catch (err) {
  connectionError = err.message;
  connectionResult = { connected: false };
} finally {
  if (client) {
    try { client.disconnect(); } catch (_) {}
  }
}

const summary = {
  service: 'redis',
  mode: 'direct-package-import',
  mjs_package: '@internal/db_connection_redis',
  app_yaml_initialized: true,
  app_yaml_storage_redis: appConfig.getNested(['storage', 'redis'], {}),
  resolved_config: {
    host: cfg.host,
    port: cfg.port,
    db: cfg.db,
    useSsl: cfg.useSsl,
  },
  connection_test: {
    ...connectionResult,
    error: connectionError,
  },
  uses_app_yaml_directly: false,
  note: 'db_connection_redis (mjs) resolves env/config/defaults; AppYamlConfig is not read by package internals.',
};

console.log(JSON.stringify(summary, null, 2));

File: platform/test/integration/storage/redis/test_case.py
#!/usr/bin/env python3
"""Direct package test for db_connection_redis with connection check."""

from __future__ import annotations

import argparse
import asyncio
import json

import re
import sys
from pathlib import Path

_HERE = Path(__file__).resolve().parent
_PLATFORM = _HERE
while _PLATFORM.name != "platform" and _PLATFORM != _PLATFORM.parent:
    _PLATFORM = _PLATFORM.parent

def _inject_paths():
    """Add polyglot/packages_py source dirs to sys.path from platform pyproject.toml."""
    toml = _PLATFORM / "pyproject.toml"
    if not toml.exists():
        return
    for line in toml.read_text().splitlines():
        m = re.search(r'path\s*=\s*"([^"]+)"', line)
        if not m:
            continue
        pkg_path = (_PLATFORM / m.group(1)).resolve()
        src = pkg_path / "src"
        sys.path.insert(0, str(src if src.is_dir() else pkg_path))

_inject_paths()

from app_yaml_load import load_app_yaml_config
from db_connection_redis import RedisConfig
from db_connection_redis.client import check_connection_status

async def main() -> int:
    parser = argparse.ArgumentParser()
    parser.add_argument("--app-env", default="dev")
    parser.add_argument("--config-dir", required=True)
    args = parser.parse_args()

    app_cfg = load_app_yaml_config(config_dir=args.config_dir, app_env=args.app_env).config
    yaml_redis = app_cfg.get_nested("storage", "redis", default={})
    redis_cfg = RedisConfig()

    connection_test = await check_connection_status(redis_cfg)

    summary = {
        "service": "redis",
        "mode": "direct-package-import",
        "python_package": "db_connection_redis",
        "app_yaml_initialized": True,
        "app_yaml_storage_redis": yaml_redis,
        "resolved_config": {
            "host": redis_cfg.host,
            "port": redis_cfg.port,
            "db": redis_cfg.db,
            "use_ssl": redis_cfg.use_ssl,
        },
        "connection_test": connection_test,
        "uses_app_yaml_directly": True,
        "note": "db_connection_redis reads AppYamlConfig.storage.redis when initialized, then falls back to env/defaults.",
    }
    print(json.dumps(summary, indent=2))
    return 0

if __name__ == "__main__":
    raise SystemExit(asyncio.run(main()))
